packages/solana/src/client.ts

```diff
@@ -394,38 +406,98 @@ export class Web3Modal extends Web3ModalScaffold {
         provider.chains.find(chain => chain.chainId === SolStoreUtil.state.currentChain?.chainId) ||
         provider.chains[0]
 
-      if (connectionChain) {
-        await this.switchNetwork(
-          SolHelpersUtil.getChainFromCaip(this.chains, `solana:${connectionChain.chainId}`)
-        )
-      } else {
+      if (!connectionChain) {
         provider.disconnect()
         throw new Error('The wallet does not support any of the required chains')
       }
 
-      SolStoreUtil.setIsConnected(true)
+      SolStoreUtil.setAddress(address)
+      await this.switchNetwork(
+        SolHelpersUtil.getChainFromCaip(this.chains, `solana:${connectionChain.chainId}`)
+      )
       SolStoreUtil.setProvider(provider)
       this.provider = provider
-      this.setAddress(address)
 
       window?.localStorage.setItem(SolConstantsUtil.WALLET_ID, provider.name)
 
       await this.setApprovedCaipNetworksData()
 
       this.watchProvider(provider)
+      SolStoreUtil.setIsConnected(true)
     } finally {
       this.setLoading(false)
     }
   }
 
   private watchProvider(provider: Provider) {
+    /*
+     * The auth RPC request handlers should be moved to the primary scaffold (Web3ModalScaffold).
+     * They are replicated in wagmi and ethers clients and the behavior should be kept the same
+     * between any client.
+     */
+
+    // eslint-disable-next-line func-style
+    const rpcRequestHandler = (request: W3mFrameTypes.RPCRequest) => {
+      if (W3mFrameHelpers.checkIfRequestExists(request)) {
+        if (!W3mFrameHelpers.checkIfRequestIsAllowed(request)) {
+          if (super.isOpen()) {
+            if (super.isTransactionStackEmpty()) {
+              return
+            }
+            if (super.isTransactionShouldReplaceView()) {
+              super.replace('ApproveTransaction')
+            } else {
+              super.redirect('ApproveTransaction')
+            }
+          } else {
+            super.open({ view: 'ApproveTransaction' })
+          }
+        }
+      } else {
+        super.open()
+        // eslint-disable-next-line no-console
+        console.error(W3mFrameRpcConstants.RPC_METHOD_NOT_ALLOWED_MESSAGE, {
+          method: request.method
+        })
+        setTimeout(() => {
+          this.showErrorMessage(W3mFrameRpcConstants.RPC_METHOD_NOT_ALLOWED_UI_MESSAGE)
+        }, 300)
+        // Provider.rejectRpcRequests()
+      }
+    }
+
+    // eslint-disable-next-line func-style
+    const rpcSuccessHandler = (_response: W3mFrameTypes.FrameEvent) => {
+      if (super.isTransactionStackEmpty()) {
+        super.close()
+      } else {
+        super.popTransactionStack()
+      }
+    }
+
+    // eslint-disable-next-line func-style
```

Comment:
any reason to disable linter?

Reply:
without it would be required to declare the function with `function name() {` which trigger ts errors about `this` and `super` as it loses the binding scope.